### PR TITLE
Adds makeshift bayonets.

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
This PR changes the way bayonets are made. This is done to treat bayonets as the legitimate military weapons they are and make them differ to what a survivor who just started in an evac shelter can do. The makeshift bayonets are made the same way normal bayonets were made before (strings and spike/blade) while making a normal bayonet would be similar to making a combat knife. This would also make finding a bayonet worth while since it would be an upgrade to the makeshift version you had before.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
You might remove the plastic handle from the bayonet and replace with wooden one.

**Comment by Noctifer-de-Mortem** [comment]
Its not so much a handle as it is a mounting point for the weapon and the bayonet and when mass producing a product it is easier to work with plastic than with wood.

File: data/json/items/gunmod/underbarrel.json [hunk 1]
```diff
@@ -415,5 +415,38 @@
     },
     "dispersion_modifier": 30,
     "flags": [ "RELOAD_ONE" ]
+  },
+  {
+    "id": "makeshift_bayonet",
+    "type": "GUNMOD",
+    "copy-from": "bayonet",
+    "name": "makeshift bayonet",
+    "description": "A makeshift version of a bayonet that consists of a mear spike with some string. It still makes a decent melee weapon providing reach attacks when attached to long arm, smg or crossbow.",
```
Comment (Mecares) [inline data/json/items/gunmod/underbarrel.json]: "mear" ?

**Comment by Rivet-the-Zombie** [comment]
Double space after periods please.

File: data/json/items/gunmod/underbarrel.json (hunk 1 above)
Comment (Zireael07) [inline data/json/items/gunmod/underbarrel.json]: Should be "mere", I assume.

File: data/json/recipes/recipe_weapon.json [hunk 2]
```diff
@@ -6828,5 +6828,130 @@
...
+},{
+  "type" : "recipe",
+  "result": "bayonet",
+  "category": "CC_WEAPON",
+  "subcategory": "CSC_WEAPON_MODS",
+  "skill_used": "fabrication",
+  "difficulty": 6,
+  "time": 360000,
+  "reversible": false,
+  "autolearn": false,
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_weapon.json]: Redundant - both reversible and autolearn are false by default.

File: data/json/recipes/recipe_weapon.json [hunk 3]
```diff
@@ -6828,5 +6828,130 @@
...
+    [
+     [
+     [ "steel_lump", 1 ],
+     [ "steel_chunk", 4 ],
+     [ "scrap", 12 ]
+     ],
+     [
+     [ "plastic_chunk", 1 ]
+     ]
+    ]
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_weapon.json]: Style should be like in newer files. A good example is `data/json/recipes/electronics/parts.json`.

**Comment by Coolthulhu** [comment]
I don't really think it's a good idea to nerf bayonets right now. They all have a big problem that weapon strikes with bayonetted guns can damage the gun (instead of the bayonet), which makes using them risky early on. Guns are also too bulky to be good melee weapons, meaning it's often better to just drop the gun and grab a spear, then pick the gun back up instead of stabbing with the gun.

**Comment by Noctifer-de-Mortem** [comment]
Coolthulhu:
> I don't really think it's a good idea to nerf bayonets right now. They all have a big problem that weapon strikes with bayonetted guns can damage the gun (instead of the bayonet), which makes using them risky early on. Guns are also too bulky to be good melee weapons, meaning it's often better to just drop the gun and grab a spear, then pick the gun back up instead of stabbing with the gun.

I don't see it as a nerf since bayonets still retain their normal stats. This just makes it so that finding a bayonet is something good since it is an upgrade to what a survivor cab make from the start. The option to make a bayonet is not taken way either it just is lower quality than its pre-cataclysmic counter-part.

Bayonetted guns do make bad melee weapons but their purpose is to quickly attack a target when there is no time to either drop or store a gun and wield a melee weapon (most likely a spear since they both have ranged attack).

As for damage to guns instead of bayonet, until that is addressed then that will be a risk survivors have to take into consideration is they are going to use a bayonet.

**Comment by BorkBorkGoesTheCode** [comment]
> I don't see it as a nerf since bayonets still retain their normal stats. This just makes it so that finding a bayonet is something good since it is an upgrade to what a survivor cab make from the start. The option to make a bayonet is not taken way either it just is lower quality than its pre-cataclysmic counter-part.

That is a nerf though, if a survivor cannot reliably access a bayonet through crafting at early level like they used to. 

I do still think the new recipes are a good thing for believability.

**Comment by Coolthulhu** [comment]
> This just makes it so that finding a bayonet is something good since it is an upgrade to what a survivor cab make from the start.

No, it makes it so that crafting a bayonet is an idea worse than before, while making pre-cataclysm bayonets just as bad as the ones we have in master.

> Bayonetted guns do make bad melee weapons but their purpose is to quickly attack a target when there is no time to either drop or store a gun and wield a melee weapon (most likely a spear since they both have ranged attack).

How often does it actually happen? I mean actually, not theoretical situations tailored specifically to make bayonets look as little bad as possible.

**Comment by Noctifer-de-Mortem** [comment]
When I play professions with guns (mostly bionic soldier) I use a bayonet when I get caught by a group of zombie and ran out of ammo fighting them. I have to dispatch them using the bayonet, kiting if possible, until I am far away to either reload or wield a melee weapon. Its not very common if you play stealthily but it is not very rare either.

**Comment by Coolthulhu** [comment]
> until I am far away to either reload or wield a melee weapon

Dropping a gun takes no time, wearing it takes just a tiny bit above 1 turn. Wielding a knife takes less than one turn. A knife - even improvised one, like crafted copper knife or a dive knife - is in most cases better than a gun with a bayonet that takes 2 turns per strike.

**Comment by BorkBorkGoesTheCode** [comment]
> Dropping a gun takes no time, wearing it takes just a tiny bit above 1 turn. Wielding a knife takes less than one turn. A knife - even improvised one, like crafted copper knife or a dive knife - is in most cases better than a gun with a bayonet that takes 2 turns per strike.

Bayonets have a low ratio of damage-time.

What about this: bayonets receive a damage buff +8, at the cost of -1 range for the non-sword bayonets.

**Comment by Noctifer-de-Mortem** [comment]
> Dropping a gun takes no time, wearing it takes just a tiny bit above 1 turn. Wielding a knife takes less than one turn. A knife - even improvised one, like crafted copper knife or a dive knife - is in most cases better than a gun with a bayonet that takes 2 turns per strike.

I was not aware dropping too no time so point taken. However, in the situation I described one turn would mean being attacked so you attack with what you have. I do agree that if you can manage a 2 tile attack you can probably drop the gun and wield a knife.

**Comment by Noctifer-de-Mortem** [comment]
> What about this: bayonets receive a damage buff +8, at the cost of -1 range for the non-sword bayonets.

Maybe then giving makeshift bayonets the original bayonet's stats and the normal bayonets have a 25% stat buff instead of the makeshift ones having 25% less than the normal ones?

**Comment by pisskop** [comment]
Do bayonets spawn with enough frequency to be a new item?

**Comment by Noctifer-de-Mortem** [comment]
Let me take a look at the spawn rates.

**Comment by pisskop** [comment]
ive found a total of like 5 of any bayonet ingame.  over my creer.

i think museums would benefit from them, so enh? antique shops arent 'mideval' but in 2050 they should be there...

weights are unfirtunately less reliable because if everything has a low weighrt than its not a big deal but if bayonets are weight 10 and rivtech caseless rounds are 70 we wont be seeing bayonets at regular spawn ratios.

**Comment by Noctifer-de-Mortem** [comment]
Military surplus store could also house them.

**Comment by mugling** [comment]
Very much against the idea of makeshift variants for the reasons @Coolthulhu outlined. We should concentrate on making the existing items actually useful

**Comment by Noctifer-de-Mortem** [comment]
This PR places bayonet spawns in more places allowing it for it to be more common. Also, the makeshift bayonets have the stats the normal bayonets had but normal ones had 25% buff. The only real change for players starting out in the shelter who want to craft a bayonet is the name of the item since the stats are the same but when you find a normal bayonet then you have an upgrade of 25%

**Comment by mugling** [comment]
Before we had one problem (bayonets are useless). This PR doesn't change that but adds a further problem (more variants of useless items).

In general this trend to add makeshift variants of everything as opposed to fixing the games mechanics or adding new ones just adds monotony.

We need to fix `REACH` weapon mods first at which point there won't be a need to add makeshift variants.

**Comment by Noctifer-de-Mortem** [comment]
This PR's goal is to address the fact that a spike attached to a gun by string is as effective as a bayonet found on a military raid, this simply makes the two a distinct thing. But if this has to wait for reach gun mods to be reworked first then I am ok with this being closed until that is done.

**Comment by mugling** [comment]
> This PR's goal is to address the fact that a spike attached to a gun by string is as effective as a bayonet found on a military raid

How do you plan to differentiate those items without making the former useless or the latter overpowered? This is the problem with having too many item variants - they all become mediocre.

**Comment by Noctifer-de-Mortem** [comment]
Since the string + spike option is to be a place holder while a bayonet is found it should be ok for it to not be as powerful as the actual bayonet. Its just something to have until you find the real thing like a stone/makeshift hammer until you find an actual hammer.

**Comment by mugling** [comment]
That doesn't answer the question:

> How do you plan to differentiate those items without making the former useless or the latter overpowered?

**Comment by Noctifer-de-Mortem** [comment]
By having the makeshift variants be a place holder until you find an actual bayonet for those who want a bayonet early on making their "uselessness" irrelevant.

**Comment by mugling** [comment]
So if your proposing a useless item with arguably marginal appeal that is definitely mod content

**Comment by Noctifer-de-Mortem** [comment]
I guess we can move many useless makeshift items (including this one) as a mod then if that seems to be the problem. If recipes can be override via json file I will do it myself.

**Comment by mugling** [comment]
Yes, you can override recipes.

**Comment by Noctifer-de-Mortem** [comment]
Guess this will be a mod then.

**Pull request closed without merging**